hazelcast/src/main/java/com/hazelcast/map/impl/MapMigrationAwareService.java

```diff
@@ -155,10 +158,16 @@ public void commitMigration(PartitionMigrationEvent event) {
         }
 
         if (SOURCE == event.getMigrationEndpoint()) {
-            removeRecordStoresHavingLesserBackupCountThan(event.getPartitionId(), event.getNewReplicaIndex());
+            removeRecordStoresHavingLesserBackupCountThan(event.getPartitionId(),
+                    event.getNewReplicaIndex());
+
+            mapServiceContext
```

Comment:
Do we need to check here if this member is still a replica before clearing the counter (same as the check is done when clearing the record stores)?

Reply:
you are right i will update PR